### PR TITLE
solve the task

### PR DESCRIPTION
**Pull request by k1llzers**

### CODE REVIEW

File: src/main/java/book/store/bookstore/models/Book.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import lombok.Data;
+
+@Entity
+@Table(name = "books")
+@Data
+public class Book {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @NotNull
```
Comment (sokimaaa) [inline src/main/java/book/store/bookstore/models/Book.java]: @NotNull is annotation for validation purposes.

You need to add constraints on database level, for this use @Column annotation.

File: src/main/java/book/store/bookstore/repositories/BookRepository.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package book.store.bookstore.repositories;
+
+import book.store.bookstore.models.Book;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface BookRepository extends JpaRepository<Book,Long> {
```
Comment (sokimaaa) [inline src/main/java/book/store/bookstore/repositories/BookRepository.java]: in this homework expected to write own implementation of BookRepository, e.g. using Hibernate
image

File: src/main/java/book/store/bookstore/repositorie/impl/BookRepositoryImpl.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        try (Session session = sessionFactory.openSession()) {
+            transaction = session.beginTransaction();
+            session.save(book);
+            transaction.commit();
+            return book;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can`t insert book into DB: " + book,e);
```
Comment (Sanderfuck) [inline src/main/java/book/store/bookstore/repositorie/impl/BookRepositoryImpl.java]: You need space after "book," before commit code use Ctrl+Alt+L and Ctrl+Alt+O for cleaning code

**Review by Sanderfuck — CHANGES_REQUESTED** [review]

File: src/main/java/book/store/bookstore/model/Book.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+@Data
+public class Book {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Column(nullable = false)
+    private String title;
+    @Column(nullable = false)
+    private String author;
+    @Column(unique = true,nullable = false)
```
Comment (koval-kn) [inline src/main/java/book/store/bookstore/model/Book.java:23]:
```suggestion
    @Column(unique = true, nullable = false)
```

**Review by koval-kn — APPROVED** [review]

**Pull request merged**